Mathlib/Data/List/SplitLengths.lean

```diff
@@ -0,0 +1,109 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Data.List.Join
+import Mathlib.Order.MinMax
+import Mathlib.Algebra.BigOperators.Group.List
+
+/-!
+# Splitting a list to chunks of specified lengths
+
+This file defines splitting a list to chunks of given lengths, and some proofs about that.
+-/
+
+variable {α : Type*} (l : List α) (sz : List ℕ)
+
+namespace List
+
+/--
+Split a list to chunks of given lengths.
+-/
+def splitLengths : List ℕ → List α → List (List α)
+  | [], _ => []
+  | n::ns, x =>
+    let (x0, x1) := x.splitAt n
+    x0 :: ns.splitLengths x1
+
+@[simp]
+theorem length_splitLengths :
+    (sz.splitLengths l).length = sz.length := by
+  induction sz generalizing l
+  · simp [splitLengths]
+  · simp [splitLengths, ‹∀ (l : List α), _›]
+
+@[simp]
+lemma splitLengths_nil : [].splitLengths l = [] := rfl
+
+@[simp]
+lemma splitLengths_cons (n : ℕ) :
+    (n :: sz).splitLengths l = l.take n :: sz.splitLengths (l.drop n) := by
+  simp [splitLengths]
+
+theorem take_splitLength (i : ℕ) : (sz.splitLengths l).take i = (sz.take i).splitLengths l := by
```

Comment:
Should this be `@[simp]` (or maybe in the other direction)?

Reply:
I don't really see either direction being simpler, so maybe it's better to leave this un-simped if there's nothing for it to simplify.